openmdao/docs/theory_manual/scaling.rst

```diff
@@ -0,0 +1,91 @@
+in this theory document.
+
+
+Scenarios: run_model
+---------------------
+
+This sequence diagram shows the simplest possible operation -- executing a single explicit component with an input that comes from an IndepVarComp
+and solved using the RunOnce solver.  It shows how the outputs and residuals vectors are placed into an unscaled state right before compute is
+called on the `Paraboloid`. You can also see where unit conversion takes places during data passing.
+
+.. figure:: sequence_diagrams/scaling_run_model.png
+   :align: center
+   :width: 100%
+   :alt: Sequence diagrams for scaling during run_model
+
+
+Scenarios: computing total derivatives with a DirectSolver
+----------------------------------------------------------
+
+This sequence diagram shows computation of derivatives using a DirectSolver for the same model as above, so now we also include the linear vectors
+which are used in the computation of derivatives. We also include the Jacobian, which is never scaled. There are two phases here during which the vectors
+are converted to a scaled state for the overall operation, and then unscaled for user calls.
+
+The first phase is linearize, where compute_partials is called on the Paraboloid. During this phase, the entire jacobian is assembled from the component
+subjacobians. Note that, while for a single iteration it might seem like a waste to scale at the start of the linearization, keep in mind that linearize
+normally follows on after execution, so the model is already in a scaled state.
+
+The second phase loops over the right hand sides and performs the LU solve. This essentially takes the contents of the linear residuals (or outputs in "rev" mode)
+and multiplies with the inverse jacobian, placing the results in the linear outputs (or residuals in "rev" mode).  This operation is always done with
+the vectors unscaled. You might ask why we don't just scale the jacobian, but that would involve n**2 multiplications compared to 2*n (or 4*n if we could eliminate the unscale
+during linearize), so for any problem larger than a few variables, it is considerably more efficient to just scale the vectors as needed.
+
+.. figure:: sequence_diagrams/scaling_compute_totals_direct.png
+   :align: center
+   :width: 100%
+   :alt: Sequence diagrams for scaling during compute_totals with a DirectSolver
+
+
+Scenarios: computing total derivatives with ScipyKrylov
+-------------------------------------------------------
+
+This sqeuence shows computation of derivatives with a Krylov iterative solver rather than a direct solver. Aside from the additional looping, the main
+difference is that unit conversion of the derivatives are also handled by unscaling the linear inputs vector when it is passed rather than being baked in
+to the assembled jacobian.
+
+.. figure:: sequence_diagrams/scaling_compute_totals_gmres.png
```

Comment:
In this diagram, it shows `ExplicitComponent` entering the unscaled state, but then `DictionaryJacobian` exiting it.

It seems like it should show `ExplicitComponent` exiting unscaled context after the `(multiply jac keys))` _[extra paren in diagram]_ and the line returning to red as it exits `ScipyKrylov.matvec`.  

There is no matching `DictionaryJacobian._unscaled_context` entry call to go with that subsequent exit call, so it looks like things just got crossed in translation between ExplicitComponent and DictionaryJacobian?

Reply:
That was a bit of a mess. Fixed.